controllers/datafoundation.go

```diff
@@ -114,3 +120,21 @@ func (r *dataFoundationReconciler) reconcileOnboardingValidationSecret() error {
 	}
 	return nil
 }
+
+func (r *dataFoundationReconciler) reconcileProviderAPIServerNetworkPolicy() error {
+	providerAPIServerNetworkPolicy := &netv1.NetworkPolicy{}
+	providerAPIServerNetworkPolicy.Name = "provider-api-server-rule"
+	providerAPIServerNetworkPolicy.Namespace = r.managedFusionOffering.Namespace
```

Comment:
I think @bindrad laid out a course where the setup of reconciled resources goes into the reconciler init function.
Let's either follow that decision or change the decision for all functions

Reply:
Adhering to  @bindrad 's changes, updated